content/courses/program-security/account-data-matching.md

```diff
@@ -427,107 +325,90 @@ pub struct SecureWithdraw<'info> {
-    await spl.mintTo(
-      connection,
-      wallet.payer,
-      mint,
-      tokenPDA,
-      wallet.payer,
-      100
-    )
+test("Secure withdraw prevents unauthorized access", async () => {
+  // Setup: Initialize vault and mint some tokens to it
+  // ... (initialization code here)
 
+  // Attempt unauthorized withdrawal
+  // Note: unauthorizedWallet represents an unauthorized wallet trying to withdraw funds
+  try {
     await program.methods
       .secureWithdraw()
       .accounts({
         vault: vaultPDA,
         tokenAccount: tokenPDA,
-        withdrawDestination: withdrawDestination,
-        authority: wallet.publicKey,
+        withdrawDestination: unauthorizedWalletWithdrawDestination,
+        authority: unauthorizedWallet.publicKey,
       })
-      .rpc()
+      .rpc();
 
-    const balance = await connection.getTokenAccountBalance(tokenPDA)
-    expect(balance.value.uiAmount).to.eq(0)
-  })
-})
-```
+    // If we reach here, the test failed
+    assert.fail("Expected an error, but withdrawal succeeded");
+  } catch (error) {
+    // Check that it's the error we expect
+    expect(error.error.errorCode.code).to.equal("ConstraintHasOne");
+  }
 
-Run `anchor test` to see that the transaction using an incorrect authority
-account will now return an Anchor Error while the transaction using correct
-accounts completes successfully.
-
-```bash
-'Program Fg6PaFpoGXkYsidMpWTK6W2BeZ7FEfcYkg476zPFsLnS invoke [1]',
-'Program log: Instruction: SecureWithdraw',
-'Program log: AnchorError caused by account: vault. Error Code: ConstraintHasOne. Error Number: 2001. Error Message: A has one constraint was violated.',
-'Program log: Left:',
-'Program log: DfLZV18rD7wCQwjYvhTFwuvLh49WSbXFeJFPQb5czifH',
-'Program log: Right:',
-'Program log: 5ovvmG5ntwUC7uhNWfirjBHbZD96fwuXDMGXiyMwPg87',
-'Program Fg6PaFpoGXkYsidMpWTK6W2BeZ7FEfcYkg476zPFsLnS consumed 10401 of 200000 compute units',
-'Program Fg6PaFpoGXkYsidMpWTK6W2BeZ7FEfcYkg476zPFsLnS failed: custom program error: 0x7d1'
+  // Check that the balance hasn't changed
+  const balance = await connection.getTokenAccountBalance(tokenPDA);
+  expect(balance.value.uiAmount).to.eq(100); // Assuming 100 tokens were initially minted
+});
+
+test("Secure withdraw allows authorized access", async () => {
+  // Perform authorized withdrawal
+  await program.methods
+    .secureWithdraw()
+    .accounts({
+      vault: vaultPDA,
+      tokenAccount: tokenPDA,
+      withdrawDestination: withdrawDestination,
+      authority: wallet.publicKey,
+    })
+    .rpc();
+
+  // Check that the withdrawal succeeded
+  const balance = await connection.getTokenAccountBalance(tokenPDA);
+  expect(balance.value.uiAmount).to.eq(0);
+});
 ```
 
-Note that Anchor specifies in the logs the account that causes the error
-(`AnchorError caused by account: vault`).
+Run `anchor test` again. You should see that the unauthorized withdrawal now
+fails, while the authorized one succeeds.
```

Comment:
I lived how the old version showed the test output. It makes the objective clearer for readers. Please re-add.